crates/driver/src/domain/competition/solution/fee.rs

```diff
@@ -204,36 +223,37 @@ impl Fulfillment {
 /// - test_adjust_quote_to_out_market_buy_order_limits
 /// - test_adjust_quote_to_in_market_sell_order_limits
 /// - test_adjust_quote_to_in_market_buy_order_limits
-fn adjust_quote_to_order_limits(
-    order_sell_amount: eth::U256,
-    order_buy_amount: eth::U256,
-    order_side: Side,
-    quote_sell_amount: eth::U256,
-    quote_buy_amount: eth::U256,
-    quote_fee_amount: eth::U256,
-) -> Result<(eth::U256, eth::U256), Error> {
-    let quote_sell_amount = quote_sell_amount
-        .checked_add(quote_fee_amount)
+pub fn adjust_quote_to_order_limits(order: Order, quote: Quote) -> Result<Amounts, Math> {
```

Comment:
Is there a more concise name for this? Maybe something like `expected_amounts` (based on whether your order is in or out of market you are expecting to get the quote or your limit price).

Also (this is not your fault, but I notice it now) referencing tests in the comments is a bit clumsy. We could simply write it as a test comment and run comment tests as part of CI.